compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs

```diff
@@ -1136,6 +1136,10 @@ pub(crate) fn codegen_intrinsic_call<'tcx>(
                 };
             ret.write_cvalue(fx, CValue::by_val(is_eq_value, ret.layout()));
         };
+
+        black_box, (c a) {
+            ret.write_cvalue(fx, a);
```

Comment:
```suggestion
            // FIXME implement black_box semantics
            ret.write_cvalue(fx, a);
```

Reply:
https://github.com/rust-lang/rust/blob/fbc5f67c8926e18f47cbf94596f61280ee7331df/compiler/rustc_codegen_cranelift/src/inline_asm.rs#L21-L22

Oh, I thought that's the desired semantics from the above lines. Should I add a FIXME there as well?

Reply:
That branch will become obsolete once this PR is merged. Strictly speaking the current implementation works fine right now as cg_clif doesn't inline the `std::hint::black_box` function, but in the future once mir inlining is enabled (or the `black_box` intrinsic gets directly exported) and cranelift gets better optimizations it will become important to change this to an actual black box.

Reply:
Should I remove that branch then?

Reply:
I can remove it after this PR.